lib/brains/forms.py

```diff
@@ -97,16 +98,23 @@ def clean_plan(self):
         self.seller_product = obj
         return data
 
+    def get_name(self, plan_id):
+        # For detailed information on descriptors see: ttp://bit.ly/1CfDiou
+        name = 'Mozilla'
+        if plan_id in products:
+            name = products[plan_id].seller.name
+            return unicode(name).replace(' ', '')
+        return name
+
     @property
     def braintree_data(self):
+        plan_id = self.seller_product.public_id
         return {
             'payment_method_token': self.cleaned_data['paymethod'].provider_id,
-            'plan_id': self.seller_product.public_id,
+            'plan_id': plan_id,
             'trial_period': False,
             'descriptor': {
-                # TODO: figure out how to get product in here
-                # https://github.com/mozilla/payments/issues/57
-                'name': 'Mozilla*product',
+                'name': self.get_name(plan_id),
```

Comment:
This has to follow a [strict format](https://developers.braintreepayments.com/javascript+python/reference/response/subscription#descriptor.name) so we should do some truncating since we're sourcing it from configured product names. It's a little hard to interpret but I think it needs to be ``Mozilla*something`` and the total needs to be no longer than 22 characters. That means we probably want to truncate the product like ``Mozilla*Concretexxxxxx`` (imagine the X's are actual letters).